### PR TITLE
Allow Get-AzLogicAppRunHistory to return more than 30 entries

### PR DESCRIPTION
**Pull request by mikejwhat**
## Description

This cmdlet currently only appears to return the first page of results from the API. Drawing inspiration from [Invoke-RestMethod](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.utility/invoke-restmethod?view=powershell-7), I have introduced FollowRelLink and MaximumFollowRelLink parameters to enable the retrieval of larger result sets.

## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
    - [x] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by isra-fel** [comment]
@mikejwhat thanks for your contribution! 😁 
Are you a Microsoft employee? If you are, please join Azure orgnization here: https://aka.ms/AzureGithub

**Comment by isra-fel** [comment]
@daviburg Hi David, could you review this enhancement to `Get-AzLogicAppRunHistory`? Thanks

**Comment by mikejwhat** [comment]
> @mikejwhat thanks for your contribution! 😁
> Are you a Microsoft employee? If you are, please join Azure orgnization here: https://aka.ms/AzureGithub

No problem! Not a Microsoft employee.

**Comment by mikejwhat** [comment]
https://github.com/Azure/azure-powershell/issues/9141

File: src/LogicApp/LogicApp/Cmdlets/LogicApp/GetAzureLogicAppRunHistoryCommand.cs [hunk 1]
```diff
@@ -55,8 +65,17 @@ public override void ExecuteCmdlet()
             base.ExecuteCmdlet();
             if (string.IsNullOrEmpty(this.RunName))
             {
-                var enumerator = LogicAppClient.GetWorkflowRuns(this.ResourceGroupName, this.Name).GetEnumerator();
-                this.WriteObject(enumerator.ToIEnumerable<WorkflowRun>(), true);
+                Page<WorkflowRun> page = LogicAppClient.GetWorkflowRuns(this.ResourceGroupName, this.Name);
```
Comment (daviburg) [inline src/LogicApp/LogicApp/Cmdlets/LogicApp/GetAzureLogicAppRunHistoryCommand.cs:68]: Please preserve the use of 'var' where possible like here

Comment (daviburg) [reply src/LogicApp/LogicApp/Cmdlets/LogicApp/GetAzureLogicAppRunHistoryCommand.cs:68]: Do keep the new variable name however as it is more descriptive of the content of the variable.

File: src/LogicApp/LogicApp/Cmdlets/LogicApp/GetAzureLogicAppRunHistoryCommand.cs [hunk 2]
```diff
@@ -55,8 +65,17 @@ public override void ExecuteCmdlet()
...
+                this.WriteObject(page.GetEnumerator().ToIEnumerable<WorkflowRun>(), true);
+
+                if (FollowRelLink && !string.IsNullOrWhiteSpace(page.NextPageLink))
```
Comment (daviburg) [inline src/LogicApp/LogicApp/Cmdlets/LogicApp/GetAzureLogicAppRunHistoryCommand.cs:71]: Also precede instance variables with this. prefix.

File: src/LogicApp/LogicApp/Cmdlets/LogicApp/GetAzureLogicAppRunHistoryCommand.cs [hunk 3]
```diff
@@ -45,6 +46,15 @@ public class AzureLogicAppRunHistoryCommand : LogicAppBaseCmdlet
         [ValidateNotNullOrEmpty]
         public string RunName { get; set; }
 
+        [Parameter(Mandatory = false, HelpMessage = "Indicates the cmdlet should follow relation links.")]
```
Comment (daviburg) [inline src/LogicApp/LogicApp/Cmdlets/LogicApp/GetAzureLogicAppRunHistoryCommand.cs:49]: Why not "next page links"? The actual property is "NextPageLink"
Same for the variable name, it should be consistent with the actual property followed.

File: src/LogicApp/LogicApp/Cmdlets/LogicApp/GetAzureLogicAppRunHistoryCommand.cs [hunk 4]
```diff
@@ -55,8 +65,17 @@ public override void ExecuteCmdlet()
...
-                this.WriteObject(enumerator.ToIEnumerable<WorkflowRun>(), true);
+                Page<WorkflowRun> page = LogicAppClient.GetWorkflowRuns(this.ResourceGroupName, this.Name);
+                this.WriteObject(page.GetEnumerator().ToIEnumerable<WorkflowRun>(), true);
+
+                if (FollowRelLink && !string.IsNullOrWhiteSpace(page.NextPageLink))
+                {
+                    for (int i = 0; i < MaximumFollowRelLink; i++)
+                    {
+                        page = LogicAppClient.GetWorkflowRuns(this.ResourceGroupName, this.Name, page.NextPageLink);
+                        this.WriteObject(page.GetEnumerator().ToIEnumerable<WorkflowRun>(), true);
```
Comment (daviburg) [inline src/LogicApp/LogicApp/Cmdlets/LogicApp/GetAzureLogicAppRunHistoryCommand.cs:76]: Best not to duplicate this block of code between first page and iteration. Consider refactoring the code to always enter and perform the first execution, while only iterating further when the next page link following is enabled.

File: src/LogicApp/LogicApp/Utilities/LogicAppClientRunOperations.cs [hunk 5]
```diff
@@ -40,9 +40,16 @@ public partial class LogicAppClient
...
         {
-            return (Page<WorkflowRun>)this.LogicManagementClient.WorkflowRuns.List(resourceGroupName, workflowName);
+            if (string.IsNullOrWhiteSpace(nextPageLink))
+            {
+                return (Page<WorkflowRun>)this.LogicManagementClient.WorkflowRuns.List(resourceGroupName, workflowName);
+            }
+            else
+            {
+                return (Page<WorkflowRun>)this.LogicManagementClient.WorkflowRuns.ListNext(nextPageLink);
+            }
```
Comment (daviburg) [inline src/LogicApp/LogicApp/Utilities/LogicAppClientRunOperations.cs:52]: Use ternary expression and a single return statement

```
return condition
    ? a
    : b;
```

**Review by daviburg — CHANGES_REQUESTED** [review]

File: src/LogicApp/LogicApp/Utilities/LogicAppClientRunOperations.cs [hunk 6]
```diff
@@ -40,9 +40,11 @@ public partial class LogicAppClient
         /// <param name="resourceGroupName">Name of the resource group</param>
         /// <param name="workflowName">Name of the workflow</param>
         /// <returns>List of workflow runs</returns>
-        public Page<WorkflowRun> GetWorkflowRuns(string resourceGroupName, string workflowName)
+        public Page<WorkflowRun> GetWorkflowRuns(string resourceGroupName, string workflowName, string nextPageLink = "")
```
Comment (daviburg) [inline src/LogicApp/LogicApp/Utilities/LogicAppClientRunOperations.cs:43]: nit: please use constant string.Empty

**Review by daviburg — APPROVED** [review]

File: src/LogicApp/LogicApp/Utilities/LogicAppClientRunOperations.cs (hunk 6 above)
Comment (mikejwhat) [reply src/LogicApp/LogicApp/Utilities/LogicAppClientRunOperations.cs:43]: Didn't think that was possible here.

Comment (daviburg) [reply src/LogicApp/LogicApp/Utilities/LogicAppClientRunOperations.cs:43]: Ah yes I forgot, it's not defined as a const but as a read-only static so we can't use it for default parameters.

Comment (mikejwhat) [reply src/LogicApp/LogicApp/Utilities/LogicAppClientRunOperations.cs:43]: Not a fan of empty quotes either

**Review by isra-fel — APPROVED** [review]
Looks good 👍

**Pull request merged**